Doc/whatsnew/3.12.rst

```diff
@@ -343,6 +343,10 @@ shutil
   will be removed in Python 3.14.
   (Contributed by Irit Katriel in :gh:`102828`.)
 
+* :func:`shutil.which` now consults the *PATHEXT* environment variable to
+  find matches within *PATH* on Windows.
```

Comment:
It always did, but now it's more consistent with regular PATH searches. That's the thing to highlight here, I believe.

Reply:
Will reword.